packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx

```diff
@@ -45,6 +69,23 @@ export const BookingFields = ({
           !!rescheduleUid &&
           bookingData !== null;
 
+        // [CAL-3992] if the name variant is firstAndLastName and Last Name is
+        // required but bookingData.responses.name has lastName empty
+        // then we should allow the user to edit the name field during reschedule
+        // since the booking owner may have updated the "required" on the lastName
+        // after the original booking was made.
+        if (
+          field.variant === "firstAndLastName" &&
+          field.variantsConfig?.variants[field.variant].fields[1].required
+        ) {
+          const nameField = bookingData?.responses.name || {};
+          // @ts-expect-error the nameField is an object and it has lastName
+          // it appears that the type definition is incorrect. TODO: Fix the type definition
+          if (nameField && Object.keys(nameField).length === 2 && !nameField.lastName) {
```

Comment:
```suggestion
          if (nameField && typeof nameField === "object" && !Array.isArray(nameField) && "lastName" in nameField) {
```

This should fix type error

Reply:
Also only allow user to edit incase the lastName is empty. If user had filled the lastname and then requirement was changed, user still would be able to edit

Reply:
> Also only allow user to edit incase the lastName is empty. If user had filled the lastname and then requirement was changed, user should NOT be able to edit

```typescript
 if (nameField && Object.keys(nameField).length === 2 && !nameField.lastName) 
```

Yes, as per my original code, that is what would happen i.e. if the user previously filled in a `lastName` and the owner updated the `lastName` to be "required" after the original booking, the fields wouldn't be editable.

Here's a video reference with my original code. Let me know if there's still a confusion.

https://github.com/calcom/cal.com/assets/19223383/0f9b4695-8d9e-4e91-a929-9a80a3c213ac

Reply:
```typescript
 if (nameField && typeof nameField === "object" && !Array.isArray(nameField) && "lastName" in nameField)
```
@Amit91848 We could narrow the type, what I meant by my TODO comment was to do it more holistically. But, if we do go ahead with type narrowing like this, we should still be checking for the following two things ` Object.keys(nameField).length === 2 && !nameField.lastName`. This is what ensures that the names aren't editable if there's already a lastName field previously.

Reply:
@Amit91848 updated this conditional logic to the following. It narrows the type, but also still works as intended.

```typescript
if (typeof nameField === "object" && "lastName" in nameField && !nameField.lastName) {
    readOnly = false;
}
```